packages/react-graphql/src/hooks/tests/mutation.test.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import gql from 'graphql-tag';
+import {createGraphQLFactory} from '@shopify/graphql-testing';
+
+import useMutation from '../mutation';
+import {
+  mountWithGraphQL,
+  prepareAsyncReactTasks,
+  teardownAsyncReactTasks,
+} from './utilities';
+
+const updatePetMutation = gql`
+  mutation UpdatePetMutation($name: String!) {
+    updatePet(name: $name) {
+      id
+    }
+  }
+`;
+
+const createGraphQL = createGraphQLFactory();
+
+const MockMutation = ({children}: {children: Function}) => {
+  const mutate = useMutation(updatePetMutation);
+  const [response, setResponse] = React.useState();
+
+  async function runMutation() {
+    const response = await mutate();
+    setResponse(response);
+  }
+
+  return (
+    <>
+      <button type="button" onClick={runMutation} />
+      {children(response)}
+    </>
+  );
+};
+
+const mockMutationData = {
+  updatePet: {id: 'new-pet-id-123', __typename: 'Cat'},
+};
+
+// This is skip because current `act` wrapper does not support async operation
+// The test will pass if we update to `react-dom` v16.9.0-alpha.0
+// and wrap `graphQL.resolveAll()` in an `act`
+// https://github.com/facebook/react/issues/14769#issuecomment-481251431
+// eslint-disable-next-line jest/no-disabled-tests
+describe.skip('useMutation', () => {
```

Comment:
feels like you are missing tests for overrideClient, options and the per-mutation options, I guess they would all just be skipped as well.